diff -Naur a/include/SDL_opengl_glext.h b/include/SDL_opengl_glext.h
--- a/include/SDL_opengl_glext.h	2019-07-25 06:32:36.000000000 +0200
+++ b/include/SDL_opengl_glext.h	2019-10-03 17:47:37.969696030 +0200
@@ -43,6 +43,15 @@
 #include <windows.h>
 #endif
 
+#ifdef __IPHONEOS__
+#include <OpenGLES/ES1/gl.h>
+#include <OpenGLES/ES1/glext.h>
+#else
+#include <GLES/gl.h>
+#include <GLES/glext.h>
+#endif
+
+
 #ifndef APIENTRY
 #define APIENTRY
 #endif
@@ -468,9 +477,6 @@
 #ifdef __MACOSX__
 typedef long GLsizeiptr;
 typedef long GLintptr;
-#else
-typedef ptrdiff_t GLsizeiptr;
-typedef ptrdiff_t GLintptr;
 #endif
 #define GL_BUFFER_SIZE                    0x8764
 #define GL_BUFFER_USAGE                   0x8765

