diff -up snes9x-1.63/vulkan/vulkan_pipeline_image.cpp.orig snes9x-1.63/vulkan/vulkan_pipeline_image.cpp
--- snes9x-1.63/vulkan/vulkan_pipeline_image.cpp.orig	2024-09-05 21:54:11.258583072 -0500
+++ snes9x-1.63/vulkan/vulkan_pipeline_image.cpp	2024-09-05 21:54:25.699710631 -0500
@@ -1,5 +1,6 @@
 #include "vulkan_pipeline_image.hpp"
 #include "slang_helpers.hpp"
+#include <cassert>
 
 namespace Vulkan
 {
@@ -263,4 +264,4 @@ void PipelineImage::create(int width, in
     framebuffer = device.createFramebufferUnique(framebuffer_create_info).value;
 }
 
-} // namespace Vulkan
\ No newline at end of file
+} // namespace Vulkan
diff -up snes9x-1.63/vulkan/vulkan_shader_chain.cpp.orig snes9x-1.63/vulkan/vulkan_shader_chain.cpp
--- snes9x-1.63/vulkan/vulkan_shader_chain.cpp.orig	2024-09-05 21:54:59.138005735 -0500
+++ snes9x-1.63/vulkan/vulkan_shader_chain.cpp	2024-09-05 21:55:20.088190445 -0500
@@ -2,6 +2,7 @@
 #include "slang_helpers.hpp"
 #include "stb_image.h"
 #include "vulkan/vulkan_enums.hpp"
+#include <cassert>
 
 namespace Vulkan
 {
@@ -597,4 +598,4 @@ bool ShaderChain::load_lookup_textures()
     return true;
 }
 
-} // namespace Vulkan
\ No newline at end of file
+} // namespace Vulkan
diff -up snes9x-1.63/vulkan/vulkan_texture.cpp.orig snes9x-1.63/vulkan/vulkan_texture.cpp
--- snes9x-1.63/vulkan/vulkan_texture.cpp.orig	2024-09-05 21:53:16.500098767 -0500
+++ snes9x-1.63/vulkan/vulkan_texture.cpp	2024-09-05 21:53:55.662445235 -0500
@@ -1,6 +1,7 @@
 #include "vulkan_texture.hpp"
 #include "vulkan/vulkan_enums.hpp"
 #include "slang_helpers.hpp"
+#include <cassert>
 
 namespace Vulkan
 {
