diff -up snes9x-1.63/gtk/src/gtk_display_driver_vulkan.cpp.orig snes9x-1.63/gtk/src/gtk_display_driver_vulkan.cpp
--- snes9x-1.63/gtk/src/gtk_display_driver_vulkan.cpp.orig	2024-09-05 21:31:49.611479717 -0500
+++ snes9x-1.63/gtk/src/gtk_display_driver_vulkan.cpp	2024-09-05 21:33:20.396144634 -0500
@@ -66,12 +66,12 @@ bool S9xVulkanDisplayDriver::init_imgui(
     init_info.Device = context->device;;
     init_info.QueueFamily = context->graphics_queue_family_index;
     init_info.Queue = context->queue;
-    init_info.DescriptorPool = imgui_descriptor_pool.get();
+    init_info.DescriptorPool = static_cast<VkDescriptorPool>(imgui_descriptor_pool.get());
     init_info.Subpass = 0;
     init_info.MinImageCount = context->swapchain->get_num_frames();
     init_info.ImageCount = context->swapchain->get_num_frames();
     init_info.MSAASamples = VK_SAMPLE_COUNT_1_BIT;
-    ImGui_ImplVulkan_Init(&init_info, context->swapchain->get_render_pass());
+    ImGui_ImplVulkan_Init(&init_info, static_cast<VkRenderPass>(context->swapchain->get_render_pass()));
 
     auto cmd = context->begin_cmd_buffer();
     ImGui_ImplVulkan_CreateFontsTexture(cmd);
@@ -253,4 +253,4 @@ void S9xVulkanDisplayDriver::save(const
 bool S9xVulkanDisplayDriver::is_ready()
 {
     return true;
-}
\ No newline at end of file
+}
